include/ppx/grfx/grfx_swapchain.h

```diff
@@ -172,12 +180,27 @@ class Swapchain
     virtual void   Destroy() override;
     friend class grfx::Device;
 
-    // Make these protected since D3D12's swapchain resize will need to call them
-    void   DestroyColorImages();
-    Result CreateDepthImages();
-    void   DestroyDepthImages();
-    Result CreateRenderPasses();
-    void   DestroyRenderPasses();
+    struct Target
+    {
+        uint32_t width  = 0;
+        uint32_t height = 0;
+
+        Result WrapColorImages(grfx::Device&, grfx::Format format, const std::vector<void*>& handles);
+        Result CreateColorImages(grfx::Device&, grfx::Format format, uint32_t imageCount);
+        void   DestroyColorImages(grfx::Device&);
+
+        Result WrapDepthImages(grfx::Device&, grfx::Format format, const std::vector<void*>& handles);
+        Result CreateDepthImages(grfx::Device&, grfx::Format format, uint32_t imageCount);
+        void   DestroyDepthImages(grfx::Device&);
+
+        Result CreateRenderPasses(grfx::Device&);
+        void   DestroyRenderPasses(grfx::Device&);
+
+        std::vector<grfx::ImagePtr>      depthImages;
```

Comment:
Each one is corresponding to a color image.

Reply:
I can understand multiple colors buffers so that when the GPU is writing to one, the other one can be used for present. But we never need to queue depth for present, so wondering why do we need multiple depth

Reply:
We do queue depth for present, see [src/ppx/xr_component.cpp:535] .

[src/ppx/xr_component.cpp:535]: https://github.com/google/bigwheels/tree/bc40a927267f83ee53399d69c8fe85f345b0f6d7/src/ppx/xr_component.cpp#535